sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegen.scala

```diff
@@ -396,7 +401,7 @@ case class WholeStageCodegen(child: SparkPlan) extends UnaryNode with CodegenSup
         s"""
            |$evaluateInputs
            |${code.code.trim}
-           |append(${code.value}.copy());
+           |append(${code.value}$doCopy);
```

Comment:
Is this correct? The row here is created with a `GenerateUnsafeProjection`. So should it be the same row? If so, we should copy the rows even `ctx.copyResult` is false.